llvm/lib/CodeGen/Analysis.cpp

```diff
@@ -593,9 +593,10 @@ bool llvm::attributesPermitTailCall(const Function *F, const Instruction *I,
 
   // Following attributes are completely benign as far as calling convention
   // goes, they shouldn't affect whether the call is a tail call.
-  for (const auto &Attr : {Attribute::Alignment, Attribute::Dereferenceable,
-                           Attribute::DereferenceableOrNull, Attribute::NoAlias,
-                           Attribute::NonNull, Attribute::NoUndef}) {
+  for (const auto &Attr :
+       {Attribute::Alignment, Attribute::Dereferenceable,
+        Attribute::DereferenceableOrNull, Attribute::NoAlias,
+        Attribute::NonNull, Attribute::NoUndef, Attribute::Range}) {
```

Comment:
As a follow up patch, this also should filter out nofpclass

Reply:
Really we should have something in Attributes that maintains whether the attribute is ABI or not

Reply:
Thanks @arsenm and @nikic . https://github.com/llvm/llvm-project/issues/91218 opened to track the follow up.